lib/core/proxy.c

```diff
@@ -398,6 +398,52 @@ static void copy_stats(struct rp_generator_t *self)
     self->src_req->proxy_stats.bytes_read.body = self->client->bytes_read.body;
 }
 
+static void copy_ssl_info(h2o_req_t *req, struct st_h2o_socket_ssl_t *ssl)
```

Comment:
Maybe rename to `retain_ssl_info`? This is not a generic copy function but is tailored for a particular use-case. Therefore, we can be (and it is good to be) specific.